libs/traits/src/liquidity_pools.rs

```diff
@@ -34,11 +35,26 @@ pub trait LPEncoding: Sized {
 	/// It's the identity message for composing messages with pack_with
 	fn empty() -> Self;
 
-	/// Retrieves the message proof hash, if the message is a proof type.
-	fn get_proof(&self) -> Option<Proof>;
+	/// Returns whether the message is a proof or not.
+	fn is_proof_message(&self) -> bool;
+
+	/// Retrieves the message hash, if the message is a proof type.
+	fn get_message_hash(&self) -> MessageHash;
 
 	/// Converts the message into a message proof type.
 	fn to_proof_message(&self) -> Self;
+
+	/// Creates a message used for initiating message recovery.
+	///
+	/// Hash - hash of the message that should be recovered.
+	/// Router - the address of the recovery router.
+	fn initiate_recovery_message(hash: [u8; 32], router: [u8; 32]) -> Self;
```

Comment:
Nit: Should use `MessageHash` type in signature
```suggestion
	fn initiate_recovery_message(hash: MessageHash, router: [u8; 32]) -> Self;
```

Reply:
I'll add it in - https://github.com/centrifuge/centrifuge-chain/pull/1974